common/viewer.cpp

```diff
@@ -442,30 +442,35 @@ namespace rs2
                  s.second.profile.stream_type() == RS2_STREAM_DEPTH ||
                  s.second.profile.stream_type() == RS2_STREAM_FISHEYE))
             {
+                auto profile_unique_id = s.second.profile.unique_id();
+                bool profile_found = streams_origin.find(profile_unique_id) != streams_origin.end();
```

Comment:
If we're doing the find here, why aren't we keeping the iterator so we don't have to find it again (line 450 and 455)?

Reply:
I personally don't like it when searches happen twice. I think performance is quality. It can be done while keeping the code readable. Short code is not always readable.